src/v/wasm/allocator.cc

```diff
@@ -26,36 +31,99 @@
 
 namespace wasm {
 
-heap_allocator::heap_allocator(config c) {
+heap_allocator::heap_allocator(config c)
+  : _memset_chunk_size(c.memset_chunk_size) {
     size_t page_size = ::getpagesize();
-    _max_size = ss::align_up(c.heap_memory_size, page_size);
+    _size = ss::align_up(c.heap_memory_size, page_size);
     for (size_t i = 0; i < c.num_heaps; ++i) {
-        auto buffer = ss::allocate_aligned_buffer<uint8_t>(
-          _max_size, page_size);
-        std::memset(buffer.get(), 0, _max_size);
-        _memory_pool.emplace_back(std::move(buffer), _max_size);
+        auto buffer = ss::allocate_aligned_buffer<uint8_t>(_size, page_size);
+        std::memset(buffer.get(), 0, _size);
```

Comment:
does this need the chunked memset behavior too?

Reply:
Actually the new way makes it easy to make this async - do you think we should do that? I guess it's better.

Reply:
avoiding stalls on start-up isn't as bad as after the system boots up, but could still be nice to avoid since monitoring systems don't necessarily distinguish between startup and booted. probably not a big deal either way?

Reply:
I ended up making the boot one async too.